hack/ensure-directories.sh

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+DIR="$(dirname "${BASH_SOURCE[0]}")"
+ROOT_DIR="$(realpath "${DIR}/..")"
+
+# Ensure all necessary directories exist with correct permissions
+echo "Creating directories with correct permissions..."
+
+# Function to ensure directory has correct ownership
+ensure_directory() {
+    local dir="$1"
+    local current_user=$(id -un)
+    local current_group=$(id -gn)
+    
+    # Create directory if it doesn't exist
+    mkdir -p "$dir"
+    
+    # Check ownership
+    local owner=$(stat -c '%U' "$dir" 2>/dev/null || echo "none")
```

Comment:
This doesn't work on MacOS.

```suggestion
    local owner=$(ls -ld "$dir" | awk '{print $3}' 2>/dev/null || echo "none")
```

Use ls + awk instead of stat for better compatibility